include/transmem/transmem.h

```diff
@@ -153,73 +193,78 @@ friend class GraphMLWriter;
+     * Best transformatio:@n
      * Let L1,..,Ln be the links required for the transformation from @a srcFrame to @a destFrame.@n
      * Let Eij be the j-th transformaton entry inserted at time tij in the link Li.@n
      * The best transformation is the transformation obtained at time t, were t minimizes the sum over all |t-tij| from 1 to n.@n@n
      * The resolution of t is 10 ms.
-     * @param srcFrame identifier of the source frame
-     * @param destFrame identifier of the destination frame
-     * @param tstamp returns at what time the best transformation was available
-     * @return matrix encoding the requested transformation
+     * Transmem caches the entries. That is, if no link of a certain transformation was updated, the cached transformation is returned.
+     * @param srcFrame Identifier of the source frame.
+     * @param destFrame Identifier of the destination frame.
+     * @param tstamp Returns at what time the best transformation was available.
+     * @return A StampedAndRatedTransformation object encoding the requested transformation and additional quality information.
      * @throws InvalidArgumentException
      * @throws NoSuchLinkFoundException
      */
-    QMatrix4x4 getBestLink(const FrameID &srcFrame, const FrameID &destFrame, Timestamp &tstamp) const;
+    StampedAndRatedTransformation getBestLink(const FrameID &srcFrame, const FrameID &destFrame);
 
-    QMatrix4x4 getBestLinkCached(const FrameID &srcFrame, const FrameID &destFrame, Timestamp &tstamp);
+    /**
+     * @fn void updateLinkConfidence(const FrameID &srcFrame, const FrameID &destFrame, const double &confidence)
+     * @brief Updates the link from @a srcFrame to @a destFrame with @a confidence. If there was never such a link
+     * registered before, a NoSuchLinkFoundException is thrown.
+     * @param srcFrame Identifier of the source frame.
+     * @param destFrame Identifier of the destination frame.
+     * @param confidence Confidence value.
+     * @throws NoSuchLinkFoundException
+     */
+    void updateLinkConfidence(const FrameID &srcFrame, const FrameID &destFrame, const double &confidence);
 
+    /**
+     * @fn voidAsJSON() const
+     * @brief Dumps the current state of the transmem as JSON to a file.
+     */
     void dumpAsJSON() const;
 
+    /**
+     * @fn void dumpAsGraphML() const
+     * @brief Dumps the current graph build by the different registered links as GraphML to a file.
+     */
     void dumpAsGraphML() const;
 
 protected:
 
     bool shortestPath(Path &p) const;
-
-    bool calculateBestPointInTime(Path &path, Timestamp &tStampBestPointInTime) const;
-
-    bool calculateTransformation(const Path &path, StampedTransformation &e) const;
+    bool bestLink(StampedAndRatedTransformation &stT, Path &p) const;
+    void calculateBestPointInTime(Path &path, Timestamp &bestPoint) const;
+    void calculateTransformation(const Path &path, StampedAndRatedTransformation &resultT) const;
```

Comment:
Here I would simply return the transform, rather than returning void and passing a reference. I understand that maybe you chose that for the sake of symmetry with `calculateBestPointInTime`. In that case as well, if we had more documentation, we could better decide.